lib/puppet.rb

```diff
@@ -36,6 +38,28 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    # TODO: Handle the rubygems case?
```

Comment:
Do we have enough information to handle this as part of this PR? @branan? What is the correct way to handle the Rubygems case?